packages/core/src/baseclient.ts

```diff
@@ -638,21 +638,13 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
         return evt;
       }
 
-      // If a trace context is not set on the event, we use the propagationContext set on the event to
-      // generate a trace context. If the propagationContext does not have a dynamic sampling context, we
-      // also generate one for it.
-      const { propagationContext } = evt.sdkProcessingMetadata || {};
-      const trace = evt.contexts && evt.contexts.trace;
-      if (!trace && propagationContext) {
-        const { traceId: trace_id, spanId, parentSpanId, dsc } = propagationContext as PropagationContext;
-        evt.contexts = {
-          trace: {
-            trace_id,
-            span_id: spanId,
-            parent_span_id: parentSpanId,
-          },
-          ...evt.contexts,
-        };
+      const propagationContext = {
+        ...isolationScope.getPropagationContext(),
+        ...(scope ? scope.getPropagationContext() : undefined),
+      };
+
+      if (propagationContext && (!evt.sdkProcessingMetadata || !evt.sdkProcessingMetadata.dynamicSamplingContext)) {
```

Comment:
is it correct that if `evt.sdkProcessingMetadata` _is_ set we never do anything here?

Reply:
My intention here was to run the branch when there is no DSC processing metadata on the event. I believe that is the case. Another way to write this would be `!(evt.sdkProcessingMetadata && evt.sdkProcessingMetadata.dynamicSamplingContext)` which would be `!(truthy && evt.sdkProcessingMetadata.dynamicSamplingContext)` which simplifies to `!(evt.sdkProcessingMetadata.dynamicSamplingContext)`.

Am I missing something?

Reply:
ah, sorry, I should have looked at the full file, we are actually only setting this in that block 😅 all good, nevermind!